src/rdkafka_transport.c

```diff
@@ -694,7 +824,16 @@ static int rd_kafka_transport_ssl_connect (rd_kafka_broker_t *rkb,
 		goto fail;
 #endif
 
-        rd_kafka_transport_ssl_clear_error(rktrans);
+    if (rktrans->rktrans_rkb->rkb_rk->rk_conf.debug | RD_KAFKA_DBG_SECURITY) {
+
+        rd_kafka_dbg(rktrans->rktrans_rkb->rkb_rk, SECURITY, "SSL",
+            "Registering handshake debug info");
+
+        SSL_set_ex_data(rktrans->rktrans_ssl, rd_kafka_ssl_verify_context_index, rktrans->rktrans_rkb->rkb_rk);
```

Comment:
verify_context_index is set up once per runtime, but is set multiple times, once for each rk instance here, that is not safe since an application may instantiate multiple rk clients and destroy them whenever.

Do we really need this handshake debug? What exactly does it provide? Do you have any example output?

Reply:
Its probably ok to not have this.  I will remove it.  There is not much value over openssl s_client.  If it is needed we can add it back at a later time.